src/core/lib/slice/slice_intern.cc

```diff
@@ -44,7 +44,7 @@
 using grpc_core::InternedSliceRefcount;
 
 typedef struct slice_shard {
-  gpr_mu mu;
+  grpc_core::Mutex* mu;
```

Comment:
I assume that the reason that this is dynamically allocated instead of being done inline is the style guide prohibition against global variables that are not trivially destructible?  If so, how about an alternative: change this to a direct data member, and instead change `g_shards` to be dynamically allocated, so that this entire struct can be treated as a C++ object.

Reply:
Yeah you're right and I just changed it as you said.